scripts/gen_kobject_list.py

```diff
@@ -58,27 +58,32 @@
 from collections import OrderedDict
 
 # Keys in this dictionary are structs which should be recognized as kernel
-# objects. Values should either be None, or the name of a Kconfig that
-# indicates the presence of this object's definition in case it is not
-# available in all configurations.
+# objects. Values are a tuple:
+#
+#  - The first item is None, or the name of a Kconfig that
+#    indicates the presence of this object's definition in case it is not
+#    available in all configurations.
+#
+#  - The second item is a boolean indicating whether it is permissible for
+#    the object to be located in user-accessible memory.
 
 # Regular dictionaries are ordered only with Python 3.6 and
 # above. Good summary and pointers to official documents at:
 # https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6
 kobjects = OrderedDict ([
-    ("k_mem_slab", None),
-    ("k_msgq", None),
-    ("k_mutex", None),
-    ("k_pipe", None),
-    ("k_queue", None),
-    ("k_poll_signal", None),
-    ("k_sem", None),
-    ("k_stack", None),
-    ("k_thread", None),
-    ("k_timer", None),
-    ("_k_thread_stack_element", None),
-    ("net_context", "CONFIG_NETWORKING"),
-    ("device", None),
+    ("k_mem_slab", (None, False)),
+    ("k_msgq", (None, False)),
+    ("k_mutex", (None, False)),
+    ("k_pipe", (None, False)),
+    ("k_queue", (None, False)),
+    ("k_poll_signal", (None, False)),
+    ("k_sem", (None, False)),
+    ("k_stack", (None, False)),
+    ("k_thread", (None, False)),
+    ("k_timer", (None, False)),
+    ("_k_thread_stack_element", (None, False)),
+    ("net_context", ("CONFIG_NETWORKING", False)),
+    ("device", (None, False)),
```

Comment:
Random note: that parameter tuple is going to evolve better if it's a dict or something with keys that can be read and defaulted.  That's a lot of "(None, False)" instead of things like:  {"dep":"CONFIG_xxx", "kernel_mem":True}

Reply:
yeah, I have plans to refactor this again for 1.15 when I merge the privilege stack generation code back into this script